jkarenzi/task-master-be
# Feat: Implement email integration

**What does this PR do?**

This PR implements email integration. The functionality of sending emails is used in email verification and two factor authentication.

**Description of the task to be completed**

- [x] Email integration with mailgun or any other provider
- [x] Implement email verification as an additional step after signing up. Unverified users should login but shouldn't be able to access the majority of the app's features
- [x] Implement 2FA upon login, for users who requested that additional step
- [x] Write appropriate documentation for resulting endpoints
- [x] Implement appropriate error handling and responses
- [x] Write comprehensive unit tests

**How can this be manually tested?**

- Upon signup, a new user should receive an email, asking them to verify their email
- Upon login, if 2FA is enabled, a user will receive a code on their email, which they will use to continue with the authentication process 

**Swagger documentation screenshot**
<img width="898" alt="qqqqq" src="https://github.com/jkarenzi/task-master-be/assets/109604986/8f258184-ae3e-4707-9aac-eb5d7721d15a">


**Test screenshot**

<img width="361" alt="aaaaa" src="https://github.com/jkarenzi/task-master-be/assets/109604986/53aa6476-7063-438d-b813-4a94544173a1">


**What are the relevant pivotal trackers/story id?**
- #4
